ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/views/PaywallView.kt

```diff
@@ -88,24 +97,29 @@ class PaywallView : FrameLayout {
      * Sets the offering id to be used to display the Paywall. If not set, the default one will be used.
      */
     fun setOfferingId(offeringId: String?) {
-        this.offeringId = offeringId
-        invalidate()
+        val offeringSelection = if (offeringId == null) {
+            OfferingSelection.None
+        } else {
+            OfferingSelection.OfferingId(offeringId)
+        }
+        paywallOptionsState.value = paywallOptionsState.value.copy(offeringSelection = offeringSelection)
     }
 
     private fun init(context: Context, attrs: AttributeSet?) {
         parseAttributes(context, attrs)
+        paywallOptionsState.value = PaywallOptions.Builder { dismissHandler?.invoke() }
+            .setListener(internalListener)
+            .setFontProvider(initialFontProvider)
+            .setOfferingId(initialOfferingId)
+            .setShouldDisplayDismissButton(shouldDisplayDismissButton ?: false)
+            .build()
         addView(
             ComposeView(context).apply {
                 setContent {
-                    val paywallOptions = PaywallOptions.Builder { dismissHandler?.invoke() }
-                        .setListener(internalListener)
-                        .setFontProvider(fontProvider)
-                        .setOfferingId(offeringId)
-                        .setShouldDisplayDismissButton(shouldDisplayDismissButton ?: false)
-                        .build()
-                    Paywall(
-                        options = paywallOptions,
-                    )
+                    val paywallOptions by remember {
+                        paywallOptionsState
```

Comment:
Hi @Jjastiny, I believe the changes in this PR would update the state every time the `setOfferingId` method is called, even if it's called multiple times. Can you explain why you think why this would only work once?

Reply:
@tonidero 
hhm yeah, I stand corrected. It does look like it's recomposing for me. Sorry for the misunderstanding!